src/main/java/com/cairn/waypoint/dashboard/service/data/HomeworkQuestionDataService.java

```diff
@@ -1,19 +1,31 @@
 package com.cairn.waypoint.dashboard.service.data;
 
-import com.cairn.waypoint.dashboard.entity.HomeworkQuestion;
-import com.cairn.waypoint.dashboard.entity.ProtocolTemplate;
+import com.cairn.waypoint.dashboard.entity.*;
 import com.cairn.waypoint.dashboard.repository.HomeworkQuestionRepository;
+import com.cairn.waypoint.dashboard.repository.ProtocolTemplateRepository;
+import com.cairn.waypoint.dashboard.repository.HomeworkQuestionLinkedProtocolTemplatesRepository;
 import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
+
+import jakarta.persistence.EntityNotFoundException;
 import org.springframework.stereotype.Service;
 
 @Service
 public class HomeworkQuestionDataService {
 
   private final HomeworkQuestionRepository homeworkQuestionRepository;
+  private final HomeworkQuestionLinkedProtocolTemplatesRepository homeworkQuestionLinkedProtocolTemplatesRepository;
+  private final ProtocolTemplateRepository protocolTemplateRepository;
+  private final ProtocolDataService protocolDataService;
```

Comment:
One must be careful not to overuse other services within a service. If there is common functionality, lean of the rule of 3 and copy said functionality until there are three instance. Then you build a `Base**Service` to share that functionality. Using one service method within another service could result in unforeseen changes when one domain changes. You're introducing a coupling between the domains/verticals that could make maintainability a problem going forward